ansible_collections/arista/avd/plugins/plugin_utils/cv_client/workflows/verify_devices_on_cv.py

```diff
@@ -89,23 +156,27 @@ async def verify_devices_in_cloudvision_inventory(
         device.serial_number = found_device_dict_by_hostname[device.hostname].key.device_id
         device.system_mac_address = found_device_dict_by_hostname[device.hostname].system_mac_address
 
-    # Now we know which devices are on CV, so we can dig deeper and check for them in I&T Studio
-    # If a device is found, we will ensure hostname is correct and if not, update the hostname.
-    # If a device is not found, we will set _exist_on_cv back to False.
-    existing_devices = [device for device in devices if device._exists_on_cv]
-    # Using set to only include a device once.
-    existing_device_tuples = set((device.serial_number, device.system_mac_address, device.hostname) for device in existing_devices)
+        # Now we know which devices are on CV, so we can dig deeper and check for them in I&T Studio
+        # If a device is found, we will ensure hostname is correct and if not, update the hostname.
+        # If a device is not found, we will set _exist_on_cv back to False.
+        existing_devices = [device for device in devices if device._exists_on_cv]
+        # Using set to only include a device once.
+        existing_device_tuples = set((device.serial_number, device.system_mac_address, device.hostname) for device in existing_devices)
 
-    LOGGER.info(
-        "verify_devices_in_cloudvision_inventory: %s existing device objects for %s unique devices in inventory",
-        len(existing_devices),
-        len(existing_device_tuples),
-    )
+        LOGGER.info(
+            "verify_devices_in_cloudvision_inventory: %s existing device objects for %s unique devices in inventory",
+            len(existing_devices),
+            len(existing_device_tuples),
+        )
+
+        if missing_devices := [device for device in devices if not device._exists_on_cv]:
+            warnings.append(missing_devices_handler(missing_devices, skip_missing_devices, "CloudVision Device Inventory"))
 
-    if missing_devices := [device for device in devices if not device._exists_on_cv]:
-        warnings.append(missing_devices_handler(missing_devices, skip_missing_devices, "CloudVision Device Inventory"))
+        return existing_devices
```

Comment:
Please remember to revert the indentation of this section. You only reverted the first part.